src/excited_workflow/monthly_nee.py

```diff
@@ -0,0 +1,97 @@
+"""Create NEE datasets from CarbonTracker model."""
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+import xarray as xr
+from dask.distributed import Client
+from onnxruntime import InferenceSession
+
+
+client = Client()
+
+
+def run_model(model_dir: Path, df: pd.DataFrame, x_keys: list[str]) -> Any:
+    """Open model and run it.
+
+    Args:
+        model_dir: path to model directory.
+        df: input dataframe for running model.
+        x_keys: list of variables required for model.
+
+    Returns:
+        Array of predictions.
+    """
+    with open(model_dir / "lightgbm.onnx", "rb") as f:
+        model = f.read()
+
+    sess = InferenceSession(model)
+    predictions_onnx = sess.run(None, {"X": df[x_keys].to_numpy()})[0]
+
+    return predictions_onnx
+
+
+def get_predictions(
+    ds_input: xr.Dataset, x_keys: list[str], ds_regions: xr.Dataset, model_dir: Path
+) -> list[pd.DataFrame]:
+    """Create NEE monthly dataset.
+
+    Args:
+        ds_input: input dataset.
+        x_keys: list of variables required for model.
+        ds_regions: dataset of regions.
+        model_dir: path to model directory.
+    """
+    dsx = ds_input[x_keys]
+    ds_merge = xr.merge([dsx, ds_regions["transcom_regions"]])
+    allnan = ds_merge.isnull().all(dim=["latitude", "longitude"]).compute()
+
+    dfs = []
+    for idx, _dtime in enumerate(ds_merge["time"]):
+        ds_sel = ds_merge.isel(time=[idx])
+        if not any([allnan.isel(time=idx)[var] for var in allnan.data_vars]):
+            ds_sel = ds_sel.compute()
+            ds_na = ds_sel.where(ds_merge["transcom_regions"] == 2)
```

Comment:
It might actually be a lot more efficient to move this step (line 55) to the top, after line 47.

Reply:
I tried all the different configuration but leaving it as is seems to be faster for some reason.